### PR TITLE
feat: implement SetupPreferences to allow usage without Xcode / Android SDK

### PR DESCRIPTION
**Pull request by okwasniewski**
This PR addresses this feature request: #71 and makes MiniSim available for both Android and iOS native developers! 

CleanShot 2023-10-10 at 12 16 19@2x

### CODE REVIEW

File: MiniSim/Menu.swift [hunk 1]
```diff
@@ -34,14 +34,18 @@ class Menu: NSMenu {
     }
     
     func getDevices() {
-        if UserDefaults.standard.androidHome == nil {
+        if UserDefaults.standard.enableAndroidEmulators && UserDefaults.standard.androidHome == nil {
```
Comment (Garfeild) [inline MiniSim/Menu.swift:37]: I suppose you can completely remove this check and update line 46 to include `UserDefaults.standard.androidHome == nil`

File: MiniSim/Menu.swift [hunk 2]
```diff
@@ -34,14 +34,18 @@ class Menu: NSMenu {
...
             return
         }
         DispatchQueue.global(qos: .userInitiated).async {
             do {
                 var devicesArray: [Device] = []
-                try devicesArray.append(contentsOf: DeviceService.getAndroidDevices())
-                try devicesArray.append(contentsOf: DeviceService.getIOSDevices())
+                if (UserDefaults.standard.enableiOSSimulators) {
```
Comment (Garfeild) [inline MiniSim/Menu.swift:43]: You don't need to use `()` in if statement.

**Review by Garfeild — COMMENTED** [review]
Hey! I hope you don't mind of me chiming in. I was looking for Hacktoberfest projects and stumbled upon this one. Couple of comments regarding these change, I hope useful ones.

**Comment by okwasniewski** [comment]
@Garfeild 

Thanks for chiming in! I always appreciate another pair of eyes looking at a PR 👍🏻

**Review by Garfeild — APPROVED** [review]
Looks good now!

**Pull request merged**